keywrap/pkcs11/keywrapper_pkcs11_test.go

```diff
@@ -0,0 +1,138 @@
+/*
+   Copyright The ocicrypt Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package pkcs11
+
+import (
+	"os"
+	"testing"
+
+	"github.com/containers/ocicrypt/config"
+	"github.com/containers/ocicrypt/crypto/pkcs11"
+	"github.com/containers/ocicrypt/utils/softhsm"
+)
+
+var (
+	SOFTHSM_SETUP = "../../scripts/softhsm_setup"
```

Comment:
i was hoping there would be a good way to encapsulate this between the variable in the test file, and the package used to invoke the softHSM.. if this is only used for the test, could we set an ENV variable. And add a check for the env variable, if it doesn't exist, then skip the tests. (since it could be triggered by someone else using the package, that way we can enforce the non-parallel rule, else they would see tests in their repo as failing). So maybe check `SoftHSMPath`, if not set, skip test, if not use it as `softhsmSetup` path.

Reply:
This variable needs point to the path relative to the location where the test is located at so that `softhsm_setup` can be found.

Reply:
I was thinking that the tests could be run with `SOFTHSM_PATH=scripts/... go test -p 1 ...`. And check if `os.Env` for `SOFTHSM_PATH` is set, if it isn't, just skip the test.

Reply:
Is `SoftHsmPath` an environment variable? The issue is that when tests run in parallel the softhsm_setup program must not do teardown while others are using the softhsm, otherwise the access to the softhsm will start failing for some test cases that are still using it. I don't see how an ENV variable helps here.

Reply:
i was thinking the ENV variable would act as a switch on/off for the test. Not sure if there's a way to check the `-p` option from the test.
```
        t.Skip("SOFTHSM_PATH needs to be set to repo /script/softhsm.sh path and tests needs to be run in serial - go test -p 1")
```

Reply:
So `SOFTHSM_PATH` would point to `scripts/softhsm_setup`? And this is to skip over tests if softhsm package of the distro wasn't installed ?

Reply:
Currently test cases are failing with this type of error message if softhsm is not installed:

```
=== RUN   TestKeyWrapPkcs11Success
    TestKeyWrapPkcs11Success: keywrapper_pkcs11_test.go:114: ../../scripts/softhsm_setup setup failed: Need softhsm2-util from softhsm2 package
        : exit status 77
--- FAIL: TestKeyWrapPkcs11Success (0.01s)
```

I think it's fair to have such a dependency for pkcs11 support and fail a **test case** if the dependency is not installed. We don't need it when we integrate the library into another project.